### PR TITLE
#2792, #2808, #2817, #2826

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#2826 When no R present, show option to connect to remote
 - show WS window when user says NO to install MRC

#2808 Clicking on library link in package manager in remote case
 - make plain text when session is remote. Add check for folder existence

#2792 Microsoft.Data.ConnectionUI binaries are unsigned
- remove binaries since VS ships them and install even without SQL tools

#2817 Different prompt for remote 

image

### CODE REVIEW

**Comment by int19h** [comment]
Unicode has a nifty cloud symbol: ☁- maybe use that instead of *?

**Comment by MikhailArkhipov** [comment]
I tried it looks extremely ugly in the font we have. Doesn't look like this nice cloud at all.

**Comment by int19h** [comment]
I just pasted it into REPL and I see what you mean :) too bad.

**Comment by int19h** [comment]
👍

**Comment by MikhailArkhipov** [comment]
I tried 2601 ☁ CLOUD and 2605 ★ BLACK STAR and 26AB ⚫ MEDIUM BLACK CIRCLE. Technically we can make this customizable.

File: src/Host/Client/Impl/Session/RSession.cs [hunk 1]
```diff
@@ -100,6 +102,13 @@ internal sealed class RSession : IRSession, IRCallbacks {
             _afterHostStartedTask = TaskUtilities.CreateCanceled(new RHostDisconnectedException());
         }
 
+        private string GetDefaultPrompt(string requestedPrompt = null) {
```
Comment (AlexanderSher) [inline src/Host/Client/Impl/Session/RSession.cs]: `GetPrompt` probably?

File: src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs [hunk 2]
```diff
@@ -216,6 +225,8 @@ internal class ConnectionManager : IConnectionManager {
         }
 
         private Dictionary<string, IConnection> CreateConnectionList() {
+            _shell.AssertIsOnMainThread();
```
Comment (AlexanderSher) [inline src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs]: `CreateConnectionList` is called from `ConnectionManager` constructor which is accessed from `RInteractiveWorkflow` constructor. There is no guarantee that it is called from UI thread, so this assert will eventually fail.

Comment (MikhailArkhipov) [inline src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs]: OK. The code was doing ShowMessage and such so it had to be on UI thread.

**Pull request merged**